src/GraphQL/Resolver/DomainContentResolver.php

```diff
@@ -136,24 +136,23 @@ public function resolveDomainFieldValue(Content $content, $fieldDefinitionIdenti
 
     public function resolveDomainRelationFieldValue(Field $field, $multiple = false)
     {
-        if (!$field->value instanceof FieldType\RelationList\Value) {
-            throw new UserError("$field->fieldTypeIdentifier is not a RelationList field value");
+        if ($field->value instanceof FieldType\RelationList\Value) {
```

Comment:
Should it also work with ImageAsset fieldtype? It's also a Relation field if I'm not mistaken.

Also I'd extract the essence from this method and split into multiple methods for Relation, RelationList, ImageAssets to avoid nasty `if` statements.

Reply:
About Image Asset, no, not really. It does load the related asset, indeed, but does extra things, and doesn't have much in common with relations overall (see `ImageAssetFieldResolver`).

I'll see what I can do to make it more readable.

Reply:
There. I have moved the ID extraction to its own method. I could break it down into two, one for Relation, one for RelationList, but imho it is good enough that way. What do you think @webhdx ?

Reply:
It's alright :+1: